runtimes/protocol/identity-management.ts

```diff
@@ -151,12 +153,43 @@ export interface IamIdentityCenterSsoTokenSource {
     profileName: string
 }
 
+export type InProgress = 'InProgress'
+export type Complete = 'Complete'
+
+export type GetSsoTokenProgressState = InProgress | Complete
+
+export const GetSsoTokenProgressState = {
+    InProgress: 'InProgress',
+    Complete: 'Complete',
+} as const
+
+export interface GetSsoTokenProgress {
+    message?: string
+    state: GetSsoTokenProgressState
+}
+
+export const GetSsoTokenProgressType = new ProgressType<GetSsoTokenProgress>()
+
+// Use this to identify the sendProgress/onProgress call from the identity server.
+// It indicates that an SSO login is currently in progress.
+export const GetSsoTokenProgressToken = 'aws/identity/getSsoToken/progressToken'
+
+export type DeviceCodeAuthorizationFlowKind = 'DeviceCode'
+export type pkceAuthorizationFlowKind = 'PKCE'
```

Comment:
nit: Prefer Pascal-case in name and value for consistency and naming convention.